pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -4837,7 +4839,9 @@ private void internalGetReplicatedSubscriptionStatusForNonPartitionedTopic(Async
     }
 
     protected CompletableFuture<Void> internalSetSchemaCompatibilityStrategy(SchemaCompatibilityStrategy strategy) {
-        return validateTopicOperationAsync(topicName, TopicOperation.SET_SCHEMA_COMPATIBILITY_STRATEGY)
+        return validateTopicPolicyOperationAsync(topicName,
+                PolicyName.SCHEMA_COMPATIBILITY_STRATEGY,
```

Comment:
Could there be a compatibility issue here? 
In addition, please add a unit test

Reply:
No, https://github.com/apache/pulsar/pull/13297 only in the master

Reply:
> In addition, please add a unit test
Done